src/module-elasticsuite-virtual-category/Model/Rule/Condition/Product.php

```diff
@@ -82,21 +89,23 @@ public function __construct(
             $data
         );
         $this->queryFactory = $queryFactory;
+        $this->categoryRepository = $categoryRepository;
     }
 
     /**
      * Build a search query for the current rule.
      *
-     * @param array $excludedCategories Categories excluded of query building (avoid infinite recursion).
+     * @param array      $excludedCategories  Categories excluded of query building (avoid infinite recursion).
+     * @param int|string $virtualCategoryRoot Category root for Virtual Category.
      *
-     * @return QueryInterface
+     * @return QueryInterface|null
      */
-    public function getSearchQuery($excludedCategories = [])
+    public function getSearchQuery($excludedCategories = [], $virtualCategoryRoot = ''): ?QueryInterface
```

Comment:
Why an empty string as default value ? Seems weird, `null` could be better, don't you think so ?